sydent/sydent.py

```diff
@@ -184,6 +189,10 @@ def sighup(signum, stack):
             self.cfg.get('userdir', 'userdir.allowed_homeservers', '')
         ))
 
+        self.next_link_domain_whitelist = set(list_from_comma_sep_string(
```

Comment:
I'm hesitant about the use of an empty set to mean "match anything", just because it's easy to forget that it is a magic value.

```
wl = self.cfg.get('general', 'next_link.domain_whitelist')
if wl == '':
    # no whitelist, match anything
    self.next_link_domain_whitelist = None
else:
    self.next_link_domain_whitelist = {x.strip() for x in wl.split(',')}
```

(and change `if next_link_domain_whitelist` to `if next_link_domain_whitelist is None` below)